pennylane/measurements/mid_measure.py

```diff
@@ -443,7 +443,7 @@ def __mul__(self, other):
         return self._transform_bin_op(lambda a, b: a * b, other)
 
     def __rmul__(self, other):
-        return self._apply(lambda v: other * v)
+        return self._apply(lambda v: other * qml.math.cast_like(v, other))
```

Comment:
This has me thinking, why do we cast `v` like `other` and not `other` like `v`? Also, should we pre-emptively introduce `cast_like` in the other dunder methods, or just fix them as needed in the future?

Reply:
Because `other` is a numeric value and `v` is abstract until we use the processing function. I see your point, but `other` is guaranteed to be carrying interface information so I think it makes a bit more sense to use that.